chapter/related-work.md

```diff
@@ -6,6 +6,90 @@ As such, it is important to get a picture of the field of current LSP projects.
 This chapter will survey a varied range of popular language servers, compare common capabilities, and implementation approaches.
 Additionally, this part aims to recognize alternative approaches to the LSP, in the form of legacy protocols, extensible development platforms LSP extensions and the emerging Language Server Index Format.
 
+
+## Previous Approaches
+
+### IDEs
+
+Before the invention of the Language Server Protocol, extensive language support used to be provided by an IDE.
+Yet, the range of officially supported languages remained relatively small [@intellij-supported-languages].
+While integration for popular languages was common, IDE grade support for less popular ones was all but guaranteed and relied mainly on community efforts.
+In fact Eclipse[@eclipse-a-platform,eclipse-www], IntelliJ[@intelliJ], and Visual Studio[@VisualStudio], to this day the most popular IDE choices, Focus on a narrow subset of languages, historically, Java and .NET.
+Additional languages can be integrated by custom (third-party) plugins or derivations of the base platform ([@list-of-eclipse,@jetbrains-all-products]).
+Due to the proprietary nature of some of these products, plugins are not compatible between different platforms.
+Many less popular languages therefore saw redundant implementations of what is essentially the same.
+For Haskell separate efforts produced an eclipse based IDE [@haskell-ide-eclips], as well as independent IntelliJ plugins [@intellij-haskell,@HaskForce].
+Importantly, the implementers of the former reported troubles with the language barrier between Haskell and the Eclipse base written in Java.
+
+The Haskell language is an exceptional example since there is also a native Haskell IDE[@haskell-for-mac] albeit that it is available only to the MacOS operating system.
+This showcases the difficulties of language tooling and its provision.
+
+In general, developing language integrations, both as the vendor of an IDE or third-party plugin developer requires extensive resources.
+[Table @tbl:plugins-size] gives an idea of the efforts-required.
+Strikingly, since the IntelliJ platform is based on the JVM, its plugin system requires the use of JVM languages [@custom-language-support]
```

Comment:
```suggestion
Since the IntelliJ platform is based on the JVM, its plugin system requires the use of JVM languages [@custom-language-support], making it hard to reuse the code of e.g. a reference compiler or interpreter.
```